src/main/java/frc/robot/OI.java

```diff
@@ -0,0 +1,39 @@
+package frc.robot;
+
+import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
+import frc.robot.Constants.kOI;
+
+/** 
+ * Handels xbox controllers.
+*/
+public class OI {
+    private CommandXboxController driverController;
+
+    private static OI instance;
+
+    /**
+     * Makes class a singelton.
+     */
+    public static OI getInstance() {
+        if (instance == null) {
+            instance = new OI();
+        }
+        return instance;
+    }
+
+    private OI() {
+        driverController = new CommandXboxController(kOI.DRIVE_PORT);
+    }
+
+    public double getDriveTrainRotation() {
```

Comment:
my logic behind this is that it allows us to change a single constant in kOI and the axis used switches, however it might make more sense to expose a method called getJoystick(int id), and have the command provide the int based on the contents of the kOI class

Reply:
imo it makes more sense for the OI class to use the kOI constants because we do similar things with other subsystems like swerve and auto

Reply:
also
> have the command provide the int based on the contents of the kOI class

no, the command shouldn't be interacting with OI at all. im pretty sure we already had the discussion of using double suppliers (or any supplier for that reason) over controller logic (to make the commands more flexible)